src/plugins/layout.ts

```diff
@@ -1,43 +1,38 @@
-import { Action, IElementLayoutState, IEventData, ILayoutEventInfo, ILayoutRoutineInfo, ILayoutState, IMutationRoutineInfo,
-  Instrumentation, IPlugin, IShadowDomInconsistentEventState, IShadowDomMutationSummary, IShadowDomNode, LayoutRoutine,
-  NumberJson, Source } from "../../clarity";
-import { config } from "./../config";
-import { addEvent, addMultipleEvents, bind, getTimestamp, instrument } from "./../core";
-import { debug, isNumber, traverseNodeTree } from "./../utils";
+import { treeToDiscoverArray } from "../../converters/toarray";
+import { Action, CharacterDataNode, DiscoverEventType, IDiscover, IElementLayoutState, IEventInfo, IInputLayoutState, ILayoutEventData,
+   ILayoutEventInfo, ILayoutRoutineInfo, ILayoutState, IMutationRoutineInfo, InputElement, Instrumentation, IPlugin,
+   IShadowDomInconsistentEventData, IShadowDomMutationSummary, IShadowDomNode, ITextLayoutState, LayoutRoutine, NumberJson,
+   Origin} from "../../declarations/clarity";
+import { config } from "../config";
+import { addEvent, addMultipleEvents, bind, getTimestamp, instrument } from "../core";
+import { debug, isNumber, traverseNodeTree } from "../utils";
+import * as EventProvider from "./layout/eventprovider";
```

Comment:
What's the purpose of event provider? Are we trying to fix a bug we had? If this is for payload reduction.. we are getting into domain knowledge here, which we should avoid.